ocs_ci/helpers/odf_cli.py

```diff
@@ -18,23 +17,40 @@
 class ODFCLIRetriever:
     def __init__(self):
         self.semantic_version = get_semantic_ocs_version_from_config()
-        self.local_cli_path = os.path.join(config.RUN["bin_dir"], "odf-cli")
+        self.local_cli_path = os.path.join(config.RUN["bin_dir"], "odf")
+
+    def check_odf_cli_binary(self):
+        """
+        Check if the ODF CLI binary exists and is executable.
+
+        Returns:
+            bool: True if the binary exists and is executable, False otherwise.
+        """
+        return os.path.isfile(self.local_cli_path) and os.access(
+            self.local_cli_path, os.X_OK
+        )
 
     def retrieve_odf_cli_binary(self):
         """
         Download and set up the ODF-CLI binary.
 
         Raises:
             NotSupportedException: If ODF CLI is not supported on the current version or deployment.
-            AssertionError: If the CLI binary is not found or not executable.
         """
         self._validate_odf_cli_support()
 
-        image = self._get_odf_cli_image()
-        self._extract_cli_binary(image)
-        self._set_executable_permissions()
-        self._verify_cli_binary()
-        self.add_cli_to_path()
+        if not self.check_odf_cli_binary():
+            image = self._get_odf_cli_image()
+            self._extract_cli_binary(image)
+            self._set_executable_permissions()
+            self.add_cli_to_path()
+
+        if not self.check_odf_cli_binary():
+            raise RuntimeError(
+                f"Failed to retrieve and set up ODF CLI binary at {self.local_cli_path}"
+            )
+
+        log.info(f"ODF CLI binary is ready at {self.local_cli_path}")
 
     def _validate_odf_cli_support(self):
         if self.semantic_version < VERSION_4_15:
```

Comment:
Maybe we can do it with a marker so it will be skipped and not failed?

Reply:
sure